File: Algorithm/Sort/Python/BubbleSort.py
from BasicUtil import CreateRandArr , Swap


def BubbleSort(arr):
    for i in range(len(arr)-1):
        for j in range(i,len(arr)):
            if arr[i] > arr[j]:
                arr[i], arr[j] = Swap(arr[i],arr[j])            
                # arr[i] , arr[j] = arr[j] , arr[i]
    return arr

if __name__ == "__main__":
    arr = CreateRandArr()
    print(arr)
    print(BubbleSort(arr))

File: Algorithm/Sort/C++/BubbleSort.cpp
#include<iostream>
#include<vector>
using namespace std;
void swap(int *x, int *y){
    int tmp = *y;
    *x = *y;
    *y = tmp;
}


void BubbleSort(vector<int> arr){
    for(int i=0; i<arr.size;i++){
        for(int j =0; i<j ; j++){
            if(arr[i]>arr[j]){
                swap(arr[i],arr[j]);
            }
        }      
    }
}